lib/explorer/backend/lazy_series.ex

```diff
@@ -116,6 +116,7 @@ defmodule Explorer.Backend.LazySeries do
     trim: 2,
     upcase: 1,
     downcase: 1,
+    slice_string: 3,
```

Comment:
i'm unsure what to call this as slice for arrays

Reply:
Yeah, I am not sure what to call it either. And I assume that, if we want to slice lists later, we don't need a list_slice, we would simply overload this function, so we shouldn't have a `_string` suffix. It could be `slice_value` or `slice_contents` or something, but I am not sure.

Reply:
Maybe I can overload the existing slice to work when dtype is string. I'll check tomorrow

Reply:
That won't work because you may still want to slice a string series both ways. :( by list, I meant when we have a series of lists in the future!

Reply:
I'd go with `substring`, e.g. https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/substr and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring. We don't necessary _have_ to overload it for lists, right?

Reply:
If we call it substring we have the same issue if/when similar functionality is implemented for lists though right?

In that instance would it be called `sublist`?

It's a slight shame its not namespaced. Then `Series.slice` would slice the series. `Series.String.slice` would slice strings etc.

I'm not sure... I'm happy to go any way on this

Reply:
I actually think the namespacing here would not help because using the same function for doing very different things could be confusing. :( I think we have two options here:

1. `substring` as proposed by @cigrainger 

2. `string_part` (to mirror Elixir's `binary_part`) - however, `binary_part` is strict, `binary_part("elixir", 0, 13)` raises if "length" falls outside and it does not support a negative index, so I don't think it is suitable.

Well, that said, then :+1: for `substring` and we might add `sublist` and `subbinary` in the future if necessary. Can you please amend? Also please add tests for things like both start and length falling out of range. :)

Reply:
I'll get on it after work. Thanks for the input both :)